hearth/lib/hearth.rb

```diff
@@ -53,6 +53,26 @@
 require_relative 'hearth/waiters/waiter'
 require_relative 'hearth/xml'
 
+# Hearth is a low-level Ruby component library for code generated clients using
+# the Smithy modeling language.
 module Hearth
   VERSION = File.read(File.expand_path('../VERSION', __dir__)).strip
+
+  # @api private
+  @config = {}
+
+  class << self
+    # @return [Hash] Returns a hash of default configuration options shared
+    #   by all constructed clients.
+    attr_reader :config
+
+    # @param [Hash] config
+    def config=(config)
```

Comment:
While I do think global config is okay for AWS clients, I'm a little skeptical about supporting it for all generic Hearth clients - I think it could be a little bit of a foot gun if someone is using smithy-ruby clients for vastly different services.

Reply:
Maybe. Let's chat offline.

Reply:
If we do want to support generic global config - what about only supporting it in the generated SDK's namespace.  Eg:  `RailsJson.config` or `AWS::SDK::S3.config`?

Reply:
I think we will also need per-service global config and AWS global config, but I think Hearth global config is still fine here. Yes there are risks, and I understand it may seem funky, but I think it's correct to have config at every level.